app/models/sync.rb

```diff
@@ -0,0 +1,61 @@
+class Sync < ApplicationRecord
+  validates :state, presence: true, inclusion: SyncMicroMachine.valid_states
+
+  def self.in_progress?
+    where(state: SyncMicroMachine.in_progress_states).any?
+  end
+
+  def self.start
+    new(state: SyncMicroMachine::UNSTARTED).start
+  end
+
+  def start
+    machine.trigger SyncMicroMachine::START
+    StartSyncJob.perform_later id
+  end
+
+  def skip
+    machine.trigger SyncMicroMachine::SKIP
+  end
+
+  def export
+    machine.trigger SyncMicroMachine::EXPORT
+  end
+
+  def copy
+    machine.trigger SyncMicroMachine::COPY
+    FinishSyncJob.perform_later id
+  end
+
+  def finalize
+    machine.trigger SyncMicroMachine::FINALIZE
+  end
+
+  def finish
+    machine.trigger SyncMicroMachine::FINISH
+  end
+
+  # this might not be neccessary
+  def increment_uploaded_parts
+    increment! :uploaded_parts
+  end
+
+  def export_folder
+    timestamp = created_at.strftime('%F%T').gsub(/[^0-9a-z ]/i, '')
```

Comment:
Right you are! Your comment actually reminded me of a little ditty we used to do at Hashrocket - here's a post that describes the technique:

http://samuelmullen.com/2009/09/named-datetime-formats-in-ror/

Basically, it's a way of naming these formats so that you don't have to look at something so archaic to get meaning out of it. Also reduces duplication, not that we have a problem with that at the moment.

In any event, I'll get a commit in that adds the initializer.